### PR TITLE
Add basic pane notation

### PR DESCRIPTION
**Pull request by e-matteson**
This is just the original Vert, Horz, Content, and Fill. Fancy stuff like DynHeight and cursor visibility comes in a later PR.

### CODE REVIEW

File: pretty/src/pane.rs [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+where
+    T: fmt::Debug,
+{
+    fn from(e: T) -> PaneError<T> {
+        PaneError::PrettyWindow(e)
+    }
+}
+
+fn divvy(cookies: usize, demands: &[PaneSize]) -> Option<Vec<usize>> {
+    let total_fixed: usize = demands.iter().filter_map(|demand| demand.get_fixed()).sum();
```
Comment (justinpombrio) [inline pretty/src/pane.rs]: TODO: make `divvy` generic over int types

Comment (justinpombrio) [reply pretty/src/pane.rs]: and replace the two `.map()`s

File: pretty/src/pane.rs [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                let child_sizes: Vec<_> = panes.iter().map(|p| p.0).collect();
+                let total_width = usize::from(self.rect().width());
+                let widths: Vec<_> = divvy(total_width, &child_sizes)
+                    .ok_or(PaneError::ImpossibleDemands)?
+                    .into_iter()
+                    .map(|n| n as Col)
+                    .collect();
+                let style = style.or(parent_style);
+                for (rect, child_note) in self
+                    .rect()
```
Comment (justinpombrio) [inline pretty/src/pane.rs]: TODO: `let rects = ...`

File: pretty/src/pane.rs (hunk 1 above)
Comment (e-matteson) [reply pretty/src/pane.rs]: Turns out that's way more complicated than the `map`s.

**Pull request merged**